src/main/java/com/example/bookshop/controller/CartController.java

```diff
@@ -0,0 +1,62 @@
+package com.example.bookshop.controller;
+
+import com.example.bookshop.dto.cart.CartItemRequestDto;
+import com.example.bookshop.dto.cart.CreateCartItemDto;
+import com.example.bookshop.dto.cart.ShoppingCartDto;
+import com.example.bookshop.service.CartService;
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/cart")
+public class CartController {
+    private final CartService cartService;
+
+    @PreAuthorize("hasAuthority('USER')")
+    @GetMapping
+    @Operation(summary = "Get all books from the shopping cart")
+    public ShoppingCartDto findAll(Pageable pageable) {
```

Comment:
add `Authentication authentication` for each endpoint
```suggestion
    public ShoppingCartDto findAll(Authentication authentication) {
```